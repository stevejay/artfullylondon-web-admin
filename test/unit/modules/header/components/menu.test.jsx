import React from 'react'
import { shallow } from 'enzyme'

import HeaderMenu from '_src/modules/header/components/menu'

it('should render correctly', () => {
  const wrapper = shallow(
    <HeaderMenu
      label='The Label'
      items={[{ label: 'Label A', path: '/path-a' }]}
      history={{}}
    />
  )

  expect(wrapper).toMatchSnapshot()
})
