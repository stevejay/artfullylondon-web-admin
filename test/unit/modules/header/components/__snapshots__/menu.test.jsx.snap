// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="src-modules-header-components-menu__container"
>
  <HeaderButton
    ariaHaspopup={true}
    label="The Label"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
  />
  <Overlay
    container={
      Menu {
        "_getOverlayTarget": [Function],
        "context": Object {},
        "handleContainerMounted": [Function],
        "handleHeaderBlur": [Function],
        "handleHeaderButtonMounted": [Function],
        "handleHeaderClick": [Function],
        "handleHeaderFocus": [Function],
        "handleHide": [Function],
        "handleItemClick": [Function],
        "handleItemMouseDown": [Function],
        "handleItemMouseEnter": [Function],
        "handleKeyDown": [Function],
        "handleKeyPress": [Function],
        "props": Object {
          "history": Object {},
          "items": Array [
            Object {
              "label": "Label A",
              "path": "/path-a",
            },
          ],
          "label": "The Label",
        },
        "refs": Object {},
        "state": Object {
          "isOpen": false,
          "selectedIndex": null,
        },
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_element": <Menu
              history={Object {}}
              items={
                Array [
                  Object {
                    "label": "Label A",
                    "path": "/path-a",
                  },
                ]
              }
              label="The Label"
            />,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_newState": null,
            "_rendered": <div
              className="src-modules-header-components-menu__container"
            >
              <HeaderButton
                ariaHaspopup={true}
                label="The Label"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
              />
              <Overlay
                container={[Circular]}
                onHide={[Function]}
                placement="bottom"
                rootClose={true}
                shouldUpdatePosition={true}
                show={false}
                target={[Function]}
                transition={[Function]}
              >
                <ul
                  className="src-modules-header-components-menu__menu"
                  role="menu"
                >
                  <li>
                    <MenuLink
                      index={0}
                      label="Label A"
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      selected={false}
                      to="/path-a"
                    />
                  </li>
                </ul>
              </Overlay>
            </div>,
            "_rendering": false,
            "_updater": [Circular],
          },
        },
      }
    }
    onHide={[Function]}
    placement="bottom"
    rootClose={true}
    shouldUpdatePosition={true}
    show={false}
    target={[Function]}
    transition={[Function]}
  >
    <ul
      className="src-modules-header-components-menu__menu"
      role="menu"
    >
      <li
        key="Label A"
      >
        <MenuLink
          index={0}
          label="Label A"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          selected={false}
          to="/path-a"
        />
      </li>
    </ul>
  </Overlay>
</div>
`;
